stripe/resource_stripe_plan.go

```diff
@@ -45,6 +41,46 @@ func resourceStripePlan() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 			},
+			"aggregate_usage": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"billing_scheme": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "per_unit",
+			},
+			"interval_count": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			// TODO Restore once we figure out how to properly delete this data with the Go client
```

Comment:
I'd appreciate some help here as Go is not my primary language.

Reply:
I checked out your branch and tried adding metadata:

```
  metadata = {
    baz = "quux"
  }
 ``` 

In order to remove some items in metadata, we need to set discard keys a value set to empty string (`""`).  One way to do detect which keys are missing is to 

1. In `resourceStripePlanUpdate`, fetching the resource from the API again 

```
	plan, err := client.Plans.Get(d.Id(), nil)

	if err != nil {
		fmt.Errorf("can't read plan id %s", d.Id())
		return err
	}
```

2. Change how we assign `metadata` to the PlanParams we create:

```	if d.HasChange("metadata") {
		params.Metadata = unsetMissingKeys(expandMetadata(d), plan.Metadata)
	}```


3. Implement the missing function :-)

```

func unsetMissingKeys(from map[string]string, with map[string]string) map[string]string {
	for key, _ := range with {
		if _, ok := from[key]; ok {
			// key found
		} else {
			from[key] = ""
		}
	}
	return from
}
```

There might a more efficient way of doing it :-)

Tell me if that helps, I can pair/finish it up.

Reply:
This was a great idea. Rather than make an extra API call, I used [`GetChange`](https://godoc.org/github.com/hashicorp/terraform/helper/schema#ResourceData.GetChange) to get the existing keys.